src/nvim/eval.c

```diff
@@ -10054,23 +10054,68 @@ static void f_getcmdwintype(typval_T *argvars, typval_T *rettv)
   rettv->vval.v_string[0] = cmdwin_type;
 }
 
-/*
- * "getcwd()" function
- */
+/// `getcwd('s')` function
+///
+/// @pre The type of the argument `s` must be a string or nothing.
 static void f_getcwd(typval_T *argvars, typval_T *rettv)
 {
-  char_u      *cwd;
+  // Pre-condition
+  if (argvars->v_type != VAR_STRING && argvars->v_type != VAR_UNKNOWN) {
+    EMSG(_(e_invarg));
+    return;
+  }
+
+  char_u *cwd = NULL;                    // current working directory to print
+  char_u *scp = argvars->vval.v_string;  // scope
 
   rettv->v_type = VAR_STRING;
   rettv->vval.v_string = NULL;
+
   cwd = xmalloc(MAXPATHL);
-  if (os_dirname(cwd, MAXPATHL) != FAIL) {
-    rettv->vval.v_string = vim_strsave(cwd);
+  cwd[0] = '\0';  // initialize to empty string
```

Comment:
Nope, I was wrong, a few lines down:
```{c}
if (from) {
  xstrlcpy((char *)cwd, (char *)from, MAXPATHL);
}

rettv->vval.v_string = cwd;

```
If it was not initialised and `from` was an empty string or a `null` pointer we would return garbage. I'd say we keep the comment, it doesn't really hurt either.